src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Transcoding.cs

```diff
@@ -1000,7 +1023,16 @@ public static OperationStatus TranscodeToUtf8(char* pInputBuffer, int inputLengt
 
                         if (Utf16Utility.AllCharsInUInt64AreAscii(possibleNonAsciiQWord)) // all chars in first QWORD are ASCII
                         {
-                            Unsafe.WriteUnaligned<uint>(pOutputBuffer, Sse2.ConvertToUInt32(Sse2.PackUnsignedSaturate(utf16Data, utf16Data).AsUInt32()));
+                            if (AdvSimd.IsSupported)
+                            {
+                                Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16Data);
+                                Vector128<byte> source = AdvSimd.ExtractNarrowingSaturateUnsignedUpper(lower, AdvSimd.LoadVector128((short*)pOutputBuffer + Vector128<short>.Count));
+                                Unsafe.WriteUnaligned<uint>(pOutputBuffer, source.AsUInt32().ToScalar());
```

Comment:
Is intent here to write only first four bytes of `source` to `pOutputBuffer`?

Reply:
The Sse2 case is saving a uint, so yes. Do you have any concerns?

Reply:
BTW, I just merged a small change to line 1029. The second argument is `utf16Data`.

Reply:
If you want just lower 4 bytes then you don't need a call to `ExtractNarrowingSaturateUnsignedUpper` because this call populate upper 8 bytes of `source`.

In other words, this should be
```c#
Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16Data);
AdvSimd.Arm64.StoreSelectedScalar(lower.AsUInt32(), 0);
```

Reply:
Thanks. Just a couple of corrections for that snippet:
```cs
Vector64<byte> lower = AdvSimd.ExtractNarrowingSaturateUnsignedLower(utf16Data);
AdvSimd.StoreSelectedScalar((uint*)pOutputBuffer, lower.AsUInt32(), 0);
```